src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -173,6 +177,111 @@ Task<TResult> IInterleavingContext.CallSubOrchestratorWithRetryAsync<TResult>(st
+        }
+
+        async Task<DurableHttpResponse> IDurableOrchestrationContext.CallHttpAsync(DurableHttpRequest req)
+        {
+            DurableHttpResponse durableHttpResponse = await this.ScheduleDurableHttpActivityFunctionAsync(req);
+
+            HttpStatusCode currStatusCode = durableHttpResponse.StatusCode;
+
+            while (currStatusCode == HttpStatusCode.Accepted && this.Config.Options.HttpSettings.AsynchronousPatternEnabled)
+            {
+                Dictionary<string, StringValues> headersDictionary = new Dictionary<string, StringValues>(durableHttpResponse.Headers);
+                DateTime currUtcTime = this.InnerContext.CurrentUtcDateTime;
+                DateTime fireAt = default(DateTime);
+                if (headersDictionary.ContainsKey("Retry-After"))
+                {
+                    fireAt = this.InnerContext.CurrentUtcDateTime.AddSeconds(int.Parse(headersDictionary["Retry-After"]));
+                }
+                else
+                {
+                    fireAt = this.InnerContext.CurrentUtcDateTime.AddMilliseconds(this.Config.Options.HttpSettings.DefaultAsyncRequestSleepTime);
+                }
+
+                await this.InnerContext.CreateTimer(fireAt, CancellationToken.None);
+
+                DurableHttpRequest durableAsyncHttpRequest = this.CreateNewHttpRequestMessageForAsyncResponse(req);
+                durableAsyncHttpRequest = AddLocationHeaderToAsyncRequest(durableAsyncHttpRequest, durableHttpResponse);
+
+                durableHttpResponse = await this.ScheduleDurableHttpActivityFunctionAsync(durableAsyncHttpRequest);
+
+                currStatusCode = durableHttpResponse.StatusCode;
+            }
+
+            return durableHttpResponse;
+        }
+
+        private async Task<DurableHttpResponse> ScheduleDurableHttpActivityFunctionAsync(DurableHttpRequest req)
+        {
+            string serializedRequest = MessagePayloadDataConverter.HttpConverter.Serialize(req);
+            JToken durableHttpResponseJson = await this.CallDurableTaskFunctionAsync<JToken>(
+                functionName: HttpOptions.HttpTaskActivityReservedName,
+                functionType: FunctionType.Activity,
+                oneWay: false,
+                instanceId: null,
+                operation: null,
+                retryOptions: null,
+                input: serializedRequest);
+
+            JsonSerializer serializer = new JsonSerializer();
+            serializer.Converters.Add(new DurableHttpResponseJsonConverter(typeof(DurableHttpResponse)));
+            DurableHttpResponse durableHttpResponse = durableHttpResponseJson.ToObject<DurableHttpResponse>(serializer);
+            return durableHttpResponse;
+        }
+
+        private static IDictionary<string, StringValues> CreateStringValuesHeaderDictionary(IList<KeyValuePair<string, string>> headers)
+        {
+            IDictionary<string, StringValues> newHeaders = new Dictionary<string, StringValues>();
```

Comment:
Looks like you forgot to add the `StringComparer.OrdinalIgnoreCase` here. To avoid problems like this, we should refactor the code so that `DurableHttpRequest` and `DurableHttpResponse` be responsible for manipulating headers inside their own constructors. See some of my other comments related to this.